crates/ruff/src/rules/flake8_pyi/rules/custom_typevar_return.rs

```diff
@@ -0,0 +1,203 @@
+use crate::checkers::ast::Checker;
+use crate::settings::types::PythonVersion;
+use ruff_diagnostics::{Diagnostic, Violation};
+use ruff_macros::{derive_message_formats, violation};
+use ruff_python_ast as ast;
+use ruff_python_ast::{ArgWithDefault, Arguments, Decorator, Expr, Ranged, TypeParam};
+use ruff_python_semantic::analyze::visibility::{
+    is_abstract, is_classmethod, is_overload, is_staticmethod,
+};
+use ruff_python_semantic::ScopeKind;
+
+/// ## What it does
+/// Checks if certain methods define a custom `TypeVar`s for their return annotation instead of
+/// using `typing_extensions.Self`. This check currently applies for instance methods that return
+/// `self`, class methods that return an instance of `cls`, and `__new__` methods.
+///
+/// ## Why is this bad?
+/// If certain methods are annotated with a custom `TypeVar` type, and the class is subclassed,
+/// type checkers will not be able to infer the correct return type.
```

Comment:
Scanned through PRs relating to Y019 in upstream repo but found no further explanation as to why this is problematic precisely.

I assumed, similar to `NonSelfReturnType` (or `PYI034`) it's hard for type checkers to infer the type when subclassed. 

Input for a more complete explanation is welcomed.

Reply:
It's mainly just to enforce a consistent style and not have to have needles TypeVar definitions cluttering up stub files